src/server/req_modify.c

```diff
@@ -899,6 +883,34 @@ req_modifyReservation(struct batch_request *preq)
 	}
 	resc_access_perm = resc_access_perm_save; /* restore perm */
 
+	temp_dur = presv->ri_wattr[RESV_ATR_duration].at_val.at_long + presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+	
+	if ((presv->ri_alter_flags & RESV_DURATION_MODIFIED) && presv->ri_alter_etime == 0) {
+		if (!is_standing || temp_dur < next_occr_start) {
+			presv->ri_alter_flags |= RESV_START_TIME_MODIFIED;
+			presv->ri_alter_etime = presv->ri_wattr[RESV_ATR_end].at_val.at_long;
+		} else {
+			log_event(PBSEVENT_RESV, PBS_EVENTCLASS_RESV, LOG_INFO,
+				preq->rq_ind.rq_modify.rq_objname, msg_stdg_resv_occr_conflict);
+			req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
+			resv_revert_alter_times(presv);
+			return;
+		}
+	}
+	
+	if ((presv->ri_alter_flags & RESV_DURATION_MODIFIED) && presv->ri_alter_stime == 0) {
+		if (!is_standing || temp_dur < next_occr_start) {
+			presv->ri_alter_stime = presv->ri_wattr[RESV_ATR_start].at_val.at_long;
+		} else {
+				log_event(PBSEVENT_RESV, PBS_EVENTCLASS_RESV, LOG_INFO,
+					preq->rq_ind.rq_modify.rq_objname, msg_stdg_resv_occr_conflict);
+				req_reject(PBSE_STDG_RESV_OCCR_CONFLICT, 0, preq);
+				resv_revert_alter_times(presv);
+				return;
+		}
+	}
+	
+
```

Comment:
At this point, you shouldn't be changing anything with ri_alter_etime/stime/duration. This is just a check to see if the reservation will run over to the next occurrance.
The checks for changing start/end already do theirs, we just need to do the duration check.
At this point in time, the start time (new one if changed, old if not) is stored in ri_wattr.
If duration has changed, then do if (start + duration) > next_occr_start: fail, else pass.

Reply:
> At this point, you shouldn't be changing anything with ri_alter_etime/stime/duration.
But we have to store the original start time and endtime before calling on start_end_dur_wall(). If we do not then for standing reservations time for next occurrences will change too.
For example if we change end time and duration, new wall time will be calculated and set. If it's a standing reservation then the start_time will change for all occurrences. This is why I am storing stime and etime before going forward